File: solutions/c/armstrong-numbers/3/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>


bool is_armstrong_number(int candidate)
{
    int copy = candidate;
    int num_digits = 0, sum_powers = 0;
    int digits[10];
    
    while(copy)
    {
        digits[num_digits] = copy % 10;
        copy /= 10;
        num_digits++;
    }
    
    for(int i = 0; i < num_digits; i++)
        sum_powers += (int)pow(digits[i], num_digits);
    
    return sum_powers == candidate;
}

File: solutions/java/armstrong-numbers/3/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    static int pow(int base, int exponent) {
        int result = 1;
        for (int count = 0; count < exponent; count++) {
            result *= base;
        }
        return result;
    }
    boolean isArmstrongNumber(int numToCheck) {
        int sum = 0;
        String copy = String.valueOf(numToCheck);

        for (int i = 0; i < copy.length(); i++) {
            sum += pow(copy.charAt(i) - '0', copy.length());
        }

        return sum == numToCheck;
    }

}
